gateway/utils/jwt/operations.go

```diff
@@ -34,7 +34,10 @@ func (j *JWT) ParseToken(ctx context.Context, token string) (map[string]interfac
 		// check if kid belongs to a normal token with kid header
 		obj, ok := j.staticSecrets[kid.(string)]
 		if ok {
-			if obj.Alg == config.RS256Public {
+			switch obj.Alg {
+			case "":
+				obj.Alg = config.HS256
+			case config.RS256Public:
 				obj.Alg = config.RS256
```

Comment:
This will cause a problem since you are overwriting the alg of the object. I would suggest you make a new object using the values of the previous one